bach/bach/quantile.py

```diff
@@ -18,35 +19,68 @@ def calculate_quantiles(
     """
     quantiles = [q] if isinstance(q, float) else q
     quantile_results = []
+
+    partition = partition or series.group_by
+    #  BigQuery requires a window function for quantiles, window frame clause is not allowed
+    window = None
+    if is_bigquery(series.engine):
+        from bach.partitioning import Window, GroupBy
+        group_by = None
+        if partition:
+            group_by = partition if isinstance(partition, GroupBy) else partition.group_by
+        window = Window(
+            group_by_columns=list(group_by.index.values()) if group_by else [],
+            order_by=[],
+            start_boundary=None,
+            end_boundary=None,
+        )
+
     for qt in quantiles:
         if qt < 0 or qt > 1:
             raise ValueError(f'value {qt} should be between 0 and 1.')
 
-        agg_result = cast(
-            'SeriesFloat64',
-            series._derived_agg_func(
+        if is_bigquery(series.engine):
+            # BigQuery names should start with a letter or underscore. Dots are not valid
+            q_col_name = f"q_{str(qt).replace('.', '_')}"
```

Comment:
this is a bit hacky because we have to replace back the dot, but I had no better alternative for this.

Reply:
Good work around this I would say :+1: 

But can we change this to:
```suggestion
            # BigQuery names should start with a letter or underscore. Dots are not valid
            q_col_name = f"__q_{str(qt).replace('.', '_')}"
```

I think it would be nice to more consistently use `__` to start internal column names. And I think that if we touch existing code or add new code, then we should go with that naming.